File: backend/internal/client/model/product.go
package model

import "time"

type Product struct {
	ID          int64      `json:"id"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	DeletedAt   *time.Time `json:"deleted_at"`
	Name        string     `json:"name"`
	Description string     `json:"description"`
	Price       float64    `json:"price"`
	UserID      *int64     `json:"user_id"`
}

File: mobile/lib/src/infrastructure/model/product/product.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hive_flutter/hive_flutter.dart';

import '../../index.dart';

part 'product.g.dart';
part 'product.freezed.dart';

@freezed
@HiveType(typeId: 0)

/// Product Control model
class Product with _$Product {
  /// Product models instruments
  const factory Product({
    @HiveField(0) int? id,
    @HiveField(1) String? title,
    @HiveField(2) String? description,
    @HiveField(3) int? category_id,
    @HiveField(4) double? price,
    @HiveField(5) String? name,
    @HiveField(6) int? min_order_quantity,
    @HiveField(7) double? rate,
    @HiveField(8) List<FileSend>? images,
    @HiveField(9) List<FileSend>? videos,
  }) = _Product;

  /// fromJson function
  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);
}
